model/riscv_sys_regs.sail

```diff
@@ -215,6 +215,14 @@ function haveZmmul()  -> bool = true
 /* Zicond extension support */
 function haveZicond() -> bool = true
 
+function legalize_priv_level (p : priv_level) -> priv_level =
+  match (haveUsrMode(), p) {
+    (false,   _) => privLevel_to_bits(Machine),
+    (true, 0b01) => if haveSupMode() then p else privLevel_to_bits(User),
+    (true, 0b10) => privLevel_to_bits(User),
+    (true,    _) => p
+  }
```

Comment:
```suggestion
  /* 
   * When a mode value is not supported, we legalize it to the lowest
   * supported mode since that is the safest choice.
   * Note: the only valid combinations of supported modes are M, M+U, M+S+U.
   */
  match p {
    0b00 => if haveUsrMode() then p else privLevel_to_bits(Machine),
    0b01 => if haveSupMode() then p else privLevel_to_bits(User),
    0b10 => privLevel_to_bits(if haveUsrMode() then User else Supervisor),
    0b11 => p
  }
```
How about something like this, so we don't need to match a tuple?

Reply:
For 0b10 - this code will lead to picking Supervisor on a M only machine... It could be fixed by - `0b10 => privLevel_to_bits(if haveUsrMode() then User else Machine)`

For 0b01 - this code will lead to picking User on a M only machine...it could be fixed by - `0b01 => if haveSupMode() then p else if haveUsrMode() then User else Machine,`

Reply:
Yes this would work. Is this preferred? Tuples are avoided because they are slow?

Reply:
I don't think that should matter - I just found the tuple version slightly more difficult to parse in my head. But that could just be a me problem :) 

I'll defer to @jrtc27 or @Alasdair for what the preferred style would be.